doc/reference/trd-legal.md

```diff
@@ -0,0 +1,188 @@
+copyright assignment is required to contribute to Tock projects.
+
+Artifacts in the Tock project MAY include explicit copyright notices.
+Substantial updates to an artifact MAY add additional copyright notices
+to an artifact. Copyright notices MUST NOT be removed by any party except
+the original copyright holder, and removal is done solely at their discretion.
+
+For full authorship information, see the version control history.
+
+
+4. Implementation
+=================
+
+All code artifacts in Tock projects MUST include a license notice.
+
+Code artifacts MAY include copyright notice(s). Copyright notices MUST
+include a year. Newer copyright notices MUST be placed after existing
+copyright notices. If non-trivial updates are performed by an original
+copyright author, they MAY amend the year(s) indicated on their existing
+copyright statement or MAY add an additional copyright line, at their
+discretion.
+
+
+4.1 Format
+----------
+
+License and copyright information MUST appear near the top of a file.
+Generally, license and copyright information SHOULD be the first lines
+in a file, however exceptions may be made where technically necessary
+(e.g. to accommodate the shebang in a shell script).
+
+License and copyright information MUST have exactly one (1) blank line
+separating it from any other content in the file.
+
+License information MUST come before copyright information.
+There MUST NOT be any whitespace between lines in the license/copyright block.
+
+Text described in this section MAY be pre-fixed or post-fixed with
+technically necessary characters (i.e. to mark as a comment in source
+code) as appropriate.
+
+The first line of license text MUST appear exactly as-follows:
+
+> Licensed under the Apache License, Version 2.0 or the MIT License.
+
+The second line of license MUST adhere to the [SPDX](https//spdx.dev)
+specification for license description. As of this writing, it SHOULD
+appear as-follows:
+
+> SPDX-License-Identifier: Apache-2.0 OR MIT
+
+The [current (v2.3) normative rules](https://spdx.github.io/spdx-spec/v2.3/SPDX-license-expressions/)
+permit case-insensitive matches of the license identifier but require
+case-sensitive matching of the disjunction operator. To simplify
+enforcement of licensing and documentation rules, Tock dictates that
+license information MUST preserve case as-shown in the SPDX license
+list (i.e. as-presented above).
+
+Copyright lines MUST follow this pattern:
+
+> Copyright {entity} {year}([,year],[-year]).
+
+The `{entity}` field is REQUIRED and should reflect the entity wishing
+to claim copyright. The `{year}` field is REQUIRED and should reflect
+when the copyright is first established. Substantial updates in the
+future MAY indicate renewed copyright, via additional comma-separated
+years or via range syntax, at the copyright holder’s discretion. The
+initial year SHALL NOT be removed unless it is the express intent of
+the copyright holder to relinquish the initial copyright.
```

Comment:
I'm not sure that "relinquish the initial copyright" is something you can *do* in many jurisdictions. Probably better to leave this half of the sentence out.